docs/index.md

```diff
@@ -66,10 +66,203 @@ Deep equal comparison. Two values are "deep equal" if:
   * `actual` can have [symbolic properties](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol) that are missing from `expectation`
 
 
+### Matcher
+
+Match values and objects by type or or other fuzzy criteria. `samsam` ships
+with these built in matchers:
+
+#### `sinon.match.any`
+
+Matches anything.
+
+
+#### `sinon.match.defined`
+
+Requires the value to be defined.
+
+
+#### `sinon.match.truthy`
+
+Requires the value to be truthy.
+
+
+#### `sinon.match.falsy`
+
+Requires the value to be falsy.
+
+
+#### `sinon.match.bool`
+
+Requires the value to be a `Boolean`
+
+
+#### `sinon.match.number`
+
+Requires the value to be a `Number`.
+
+
+#### `sinon.match.string`
+
+Requires the value to be a `String`.
+
+
+#### `sinon.match.object`
+
+Requires the value to be an `Object`.
+
+
+#### `sinon.match.func`
+
+Requires the value to be a `Function`.
+
+
+#### `sinon.match.array`
+
+Requires the value to be an `Array`.
+
+
+#### `sinon.match.array.deepEquals(arr)`
+
+Requires an `Array` to be deep equal another one.
+
+
+#### `sinon.match.array.startsWith(arr)`
+
+Requires an `Array` to start with the same values as another one.
+
+
+#### `sinon.match.array.endsWith(arr)`
+
+Requires an `Array` to end with the same values as another one.
+
+
+#### `sinon.match.array.contains(arr)`
+
+Requires an `Array` to contain each one of the values the given array has.
+
+
+#### `sinon.match.map`
+
+Requires the value to be a `Map`.
+
+
+#### `sinon.match.map.deepEquals(map)`
+
+Requires a `Map` to be deep equal another one.
+
+#### `sinon.match.map.contains(map)`
+
+Requires a `Map` to contain each one of the items the given map has.
+
+
+#### `sinon.match.set`
+
+Requires the value to be a `Set`.
+
+
+#### `sinon.match.set.deepEquals(set)`
+
+Requires a `Set` to be deep equal another one.
+
+
+#### `sinon.match.set.contains(set)`
+
+Requires a `Set` to contain each one of the items the given set has.
+
+
+#### `sinon.match.regexp`
+
+Requires the value to be a regular expression.
+
+
+#### `sinon.match.date`
+
+Requires the value to be a `Date` object.
+
+
+#### `sinon.match.symbol`
+
+Requires the value to be a `Symbol`.
+
+#### `sinon.match.in(array)`
+
+Requires the value to be in the `array`.
+
+#### `sinon.match.same(ref)`
+
+Requires the value to strictly equal `ref`.
+
+#### `sinon.match.typeOf(type)`
```

Comment:
We can further improve `samsam` to support all the types that `type-detect` can detect: https://github.com/chaijs/type-detect#readme

Reply:
Yes, that would be very easy to do. 👍

Reply:
I wonder why we use `sinon.match` and not `samsam.match` on all of these. Seems super weird when reading the docs under the assumption that this is a standalone library.